bigtable/examples/run_examples_utils.sh

```diff
@@ -19,6 +19,47 @@ readonly CBT_CMD="${CBT:-${GOPATH}/bin/cbt}"
 readonly CBT_EMULATOR_CMD="${CBT_EMULATOR:-${GOPATH}/bin/emulator}"
 readonly CBT_INSTANCE_ADMIN_EMULATOR_CMD="../tests/instance_admin_emulator"
 
+function cleanup_instance {
+  local project=$1
+  local instance=$2
+  shift 2
+
+  if [ -z "${project}" -o -z "${instance}" ]; then
+    return
+  fi
+
+  echo
+  echo "Cleaning up test instance projects/${project}/instances/${instance}"
+  local setenv="env"
+  if [ -n "${BIGTABLE_INSTANCE_ADMIN_EMULATOR_HOST:-}" ]; then
+    setenv="env BIGTABLE_EMULATOR_HOST=${BIGTABLE_INSTANCE_ADMIN_EMULATOR_HOST}"
+  fi
+  ${setenv} ../examples/bigtable_samples_instance_admin delete-instance "${project}" "${instance}"
+}
+
+function exit_handler {
+  local project=$1
+  local instance=$2
+  shift 2
+
+  cleanup_instance "${project}" "${instance}"
+  if [ -n "${BIGTABLE_INSTANCE_ADMIN_EMULATOR_HOST:-}" ]; then
+    # If the test is running against the emulator there is no need to cleanup
```

Comment:
The code says:

```
cleanup_instance
if [ condition ]; then
   # If the test is running against the emulator there's no need to cleanup the instance
   kill_emulators
fi
```

which contradicts the comment, since in some cases, we will do `cleanup_instance` and then possibly kill emulators. If it's really only possible to have one or the other, it should probably look like:

```
if [ condition ]; then
  kill_emulators
else
  cleanup_instance
fi
```

where `cleanup_instance` non-optionally deletes the instance, and complains if project or instance is unspecified.

Reply:
Took option 2.

Reply:
and thanks!  the code is much clearer.